absl/time/CMakeLists.txt

```diff
@@ -0,0 +1,80 @@
+#
+# Copyright 2017 The Abseil Authors.
+#
+# Licensed under the Apache License Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing software
+# distributed under the License is distributed on an "AS IS" BASIS
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+list(APPEND TIME_PUBLIC_HEADERS
+  "clock.h"
+  "time.h"
+)
+
+
+list(APPEND TIME_INTERNAL_HEADERS
+  "internal/test_util.h"
+)
+
+list(APPEND TIME_SRC
+    "time.cc"
+    "clock.cc"
+    "duration.cc"
+    "format.cc"
+    ${TIME_PUBLIC_HEADERS}
+    ${TIME_INTERNAL_HEADERS}
+)
+set(TIME_PUBLIC_LIBRARIES base stacktrace int128 ${CCTZ_LIBRARIES})
+
+abseil_library(
+  TARGET
+      time
+  SOURCES
+      ${TIME_SRC}
+  PUBLIC_LIBRARIES
+      ${TIME_PUBLIC_LIBRARIES}
+)
+add_dependencies(time ${ABSEIL_PRE_DEPENDENCIES})
+
+
+install(FILES ${TIME_PUBLIC_HEADERS}
+        DESTINATION ${CMAKE_INSTALL_FULL_INCLUDEDIR}/absl/time)
+install(FILES ${TIME_INTERNAL_HEADERS}
+        DESTINATION ${CMAKE_INSTALL_FULL_INCLUDEDIR}/absl/time/internal)
+
+
+#
+## TESTS
+#
+
+# test time_test
+list(APPEND TIME_TEST_SRC
```

Comment:
I still don't see the benefit of this syntax over the following:

```cmake
abseil_test(
  TARGET
      time_test
  SOURCES
      time_test.cc
      clock_test.cc
      duration_test.cc
      format_test.cc
      time_norm_test.cc
      time_test.cc
      time_zone_test.cc
      internal/test_util.cc
  PUBLIC_LIBRARIES
      time
)
```

Why make a variables for one-time use?

Reply:
Consistency in use $FOO_SRC and similar will be nice for us in the future when we generate cmake files from build files for consistency.  Although if they're generated then re-spelling files isn't that bad either I suppose.  It doesn't matter as long as it's consistent, build files are ideally for machines, not humans